### PR TITLE
Added specific models for POST APIs

### PR DESCRIPTION
**Pull request by Gaitholabi**

### CODE REVIEW

File: src/Model/Shared/CustomField.php [hunk 1]
```diff
@@ -288,37 +310,37 @@ public function setName(?string $name): self
     }
 
     /**
-     * @return bool|null
+     * @return int|null
```
Comment (jshah4517) [inline src/Model/Shared/CustomField.php:313]: Did you intentionally change these back to `int`?

Comment (Gaitholabi) [reply src/Model/Shared/CustomField.php:313]: Nope, by mistake after copying model from another branch.

File: test/DataFixtures/ApiCalls/SelfServiceApisData.php [hunk 2]
```diff
@@ -50,7 +51,7 @@ public function postApiCalls(): array
         $commentData = new CommentData;
 
         return [
-            'postComment' => [$commentData->getFilledInstance(), $commentData->getResponse()],
+            'postComment' => [(new CreateCommentData)->getFilledInstance(), $commentData->getResponse()],
```
Comment (bytestream) [inline test/DataFixtures/ApiCalls/SelfServiceApisData.php:54]: What's going on here? 🤔 what does `$commentData->getResponse()` do?

Comment (Gaitholabi) [reply test/DataFixtures/ApiCalls/SelfServiceApisData.php:54]: the input to the test function data is a `CreateComment` object, the output is a `Comment` Object.
`getResponse` is essentially just the array data of a comment, wrapped in the API format, and that's what the API expects as an output inside the Request contents

in the data fixtures, we have:
1- `getArrayData`  which is basically the array format of any model and its child models.
2- `getResponse`, same above data, but wrapped in ApiFormat
3- `getAllResponse` same as the above, but the data is in format of array of objects
4- `getDataWithObjects` same as array data, but child models are objects not arrays
5- `getFilledInstaces` is essentially, $model->fill(getDataWithObjects());

**Review by bytestream — APPROVED** [review]

**Pull request merged**